File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.py
def solution(s):
    answer = ""
    num = ["zero", "one", "two", "three", "four", "five", "six", "seven", "eight", "nine"]
    
    num_en = ""
    for i in range(len(s)):
        if s[i].isnumeric():
            answer += str(s[i])
        else:
            num_en += str(s[i])
            if num_en in num:
                index_num = num.index(num_en)
                answer += str(index_num)
                num_en = ""
                
            
    
    return int(answer)

File: 프로그래머스/1/81301. 숫자 문자열과 영단어/숫자 문자열과 영단어.js
function solution(s) {
    var answer = ""
    let check = ""
    let number = {
        "zero": "0", 
        "one": "1",
        "two": "2",
        "three": "3",
        "four": "4",
        "five": "5",
        "six": "6",
        "seven": "7",
        "eight": "8",
        "nine": "9"
    };
    for(let i = 0; i < s.length; i++){
        if(!isNaN(s[i])) answer += String(s[i]);
        else{
            check += s[i];
            if(number[check] !== undefined){
                 answer += number[check];
                 check = "";
            }
        }
        // console.log(`s[${i}] = ${s[i]}, answer = ${answer}, check = ${check}`)
        
    }
    
    // console.log(answer)
    return Number(answer);
}